src/Sylius/Behat/Context/Setup/OrderContext.php

```diff
@@ -942,6 +954,31 @@ private function createOrders(
         $this->objectManager->flush();
     }
 
+    private function createPaidOrders($numberOfCustomers, $numberOfOrders, $total): void
+    {
+        $customers = $this->generateCustomers($numberOfCustomers);
+        $sampleProductVariant = $this->sharedStorage->get('variant');
+        $total = $this->getPriceFromString($total);
+
+        for ($i = 0; $i < $numberOfOrders; ++$i) {
+            $order = $this->createOrder($customers[random_int(0, $numberOfCustomers - 1)], '#' . uniqid());
+            $order->setState(OrderInterface::STATE_NEW); // Temporary, we should use checkout to place these orders.
+            $this->applyPaymentTransitionOnOrder($order, PaymentTransitions::TRANSITION_COMPLETE);
+
+            $price = $i === ($numberOfOrders - 1) ? $total : random_int(1, $total);
+            $total -= $price;
+
+            $this->addVariantWithPriceToOrder($order, $sampleProductVariant, $price);
+
+           $order->setPaymentState(OrderPaymentStates::STATE_PAID);
```

Comment:
Never, ever call `set*State`. It should be triggered by state machine.

Reply:
+ wrong indent 💃